src/components/SignUpProfile/SignUpProfile.jsx

```diff
@@ -0,0 +1,82 @@
+import * as React from "react"
+import { useState } from "react"
+import { Link, useNavigate } from "react-router-dom"
+import { Button, Card, Alert } from "react-bootstrap" 
+import { doc, updateDoc } from "firebase/firestore"
+import { useAuth } from "../../contexts/AuthContext"
+import { db } from "../../firebase"
+import "./SignUpProfile.css"
+
+export default function SignUpProfile(props) {
+    // get user data from the database
+    const { currentUser } = useAuth()
+    const navigate = useNavigate()
+    const [loading, setLoading] = useState(false)
+    const diets = ["Gluten Free", "Ketogenic", "Vegetarian", "Vegan", "Pescetarian", "Paleo"]
+    const [dietChecked, setDietChecked] = useState([false, false, false, false, false, false])
+    const allergies = ["Diary", "Peanut", "Soy", "Egg", "Shellfish", "Tree Nut", "Gluten"]
+    const [allergiesChecked, setAllergiesChecked] = useState([false, false, false, false, false, false, false]);
```

Comment:
This is likely to get out of sync with the array above. Consider doing something like 
```
useState(new Array(allergies.length()).fill(false))
```